video/out/wldmabuf/ra_wldmabuf.c

```diff
@@ -34,25 +36,29 @@ bool ra_compatible_format(struct ra *ra, int imgfmt, uint32_t drm_format, uint64
     struct vo_wayland_state *wl = p->vo->wl;
     struct drm_format *formats = wl->compositor_formats;
 
-
-    // If we were able to make the DRM query, filter out the GPU formats.
-    // If not, just assume they all work and hope for the best.
-    if (wl->gpu_formats) {
-        bool supported_gpu_format = false;
-        for (int i = 0; i < wl->num_gpu_formats; i++) {
-            if (drm_format == wl->gpu_formats[i]) {
-                supported_gpu_format = true;
-                break;
+    // Always check if the compositor supports the format.
+    bool supported_compositor_format = false;
+    for (int i = 0; i < wl->num_compositor_formats; ++i) {
+        if (drm_format == formats[i].format) {
+            // Only accept the format if it comes with a real modifier.
+            if (modifier == formats[i].modifier && modifier != DRM_FORMAT_MOD_INVALID) {
+                return true;
+            } else if (formats[i].modifier != DRM_FORMAT_MOD_INVALID) {
+                supported_compositor_format = true;
```

Comment:
After recent change, I think we could simplify this conditions to something like. If I'm reading it right. Just a suggestion.

```suggestion
    // Always check if the compositor supports the format.
    bool supported_compositor_format = false;
    for (int i = 0; i < wl->num_compositor_formats; ++i) {
        if (formats[i].format != drm_format || formats[i].modifier == DRM_FORMAT_MOD_INVALID)
            continue;
        if (modifier == formats[i].modifier)
            return true;
        supported_compositor_format = true;
```

Reply:
Yeah that works.